RELEASE.md

```diff
@@ -0,0 +1,304 @@
+# Release process
+
+The release process starts with one or more tagged release candidates,
+when no more blocking issues needs to be fixed a final tag is created
+and the final release is rolled.
+
+confluent-kafka-dotnet uses semver versioning and loosely follows
+librdkafka's version, e.g. v0.11.x for the final release and
+v0.11.x-RC3 for the 3rd v0.11.x release candidate.
+
+With the addition of prebuilt binary packages we make use of travis-ci.org.
```

Comment:
we don't use travis-ci for this, and the .net client has always used prebuilt binaries. Maybe change the wording to something like:

"Artifacts corresponding to each tag are built automatically by Appveyor. You can access these via: https://ci.appveyor.com/project/ConfluentClientEngineering/confluent-kafka-dotnet/history . Appveyor builds five nuget packages, which all need to be manually uploaded to nuget.org to complete the release process. It is sometimes useful to upload release candidate packages for the purposes of cross platform smoke testing, however these should be hidden following the actual release."